backend/market/serializers.py

```diff
@@ -50,46 +61,41 @@ def get_image_url(self, obj):
             return image.url
 
     class Meta:
-        model = SubletImage
+        model = ItemImage
         fields = ["id", "image_url"]
 
 
-# complex sublet serializer for use in C/U/D + getting info about a singular sublet
-class SubletSerializer(serializers.ModelSerializer):
-    # amenities = AmenitySerializer(many=True, required=False)
-    # images = SubletImageURLSerializer(many=True, required=False)
-    amenities = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=Amenity.objects.all(), required=False
+# complex item serializer for use in C/U/D + getting info about a singular tag
+class ItemSerializer(serializers.ModelSerializer):
+    # amenities = ItemSerializer(many=True, required=False)
+    images = ItemImageURLSerializer(many=True, required=False)
+
+    tags = serializers.PrimaryKeyRelatedField(many=True, queryset=Tag.objects.all(), required=False)
+    category = serializers.PrimaryKeyRelatedField(queryset=Category.objects.all(), required=True)
+    seller = serializers.PrimaryKeyRelatedField(queryset=User.objects.all(), required=False)
+    favorites = serializers.PrimaryKeyRelatedField(
+        many=True, queryset=User.objects.all(), required=False
     )
 
     class Meta:
-        model = Sublet
-        read_only_fields = [
-            "id",
-            "created_at",
-            "subletter",
-            "sublettees",
-            # "images"
-        ]
+        model = Item
+        read_only_fields = ["id", "created_at", "seller", "buyer", "images"]
```

Comment:
Buyers is initialized so that it's empty upon creation of an object. It's a `ManyToMany` field, so if it isn't passed to `ModelSerializer's` create method, `ModelSerializer` will just not update the association table, which implicitly initializes buyers to an empty set. You could double check this, I read about this in `ModelSerializer's` source code at line 1011.

Seller is manually set to the user which made the request before being added to validated_data and passed to `ModelSerializer's` create function. You can think of`read_only_field` as being read-only from the user's point of view (it won't be updated based on user input).

BTW i realized that `read_only_fields` should contain `buyers`, not `buyer`